certification/artifacts/filesystem_writer.go

```diff
@@ -0,0 +1,58 @@
+package artifacts
+
+import (
+	"fmt"
+	"io"
+	"path/filepath"
+
+	"github.com/spf13/afero"
+)
+
+// FilesystemWriter is an ArtifactWriter that targets a particular directory on
+// the underlying filesystem.
+type FilesystemWriter struct {
+	dir string
+	fs  afero.Fs
+}
+
+// NewFilesystemWriter creates an artifact writer which writes to the filesystem.
+func NewFilesystemWriter(opts ...FilesystemWriterOption) (*FilesystemWriter, error) {
```

Comment:
```suggestion
func NewFilesystemWriter(opts ...FilesystemWriterOption) *FilesystemWriter {
```

Error is never returned.

Reply:
I'm open to the idea, but if these should need to do something that might return an error in the future, it's easier to return nil now than it is to add an error return value to the signature later.

Reply:
Ah I see, okay I figured it was an oversight.  This works for me.